.github/workflows/pre-commit-ci.yml

```diff
@@ -0,0 +1,50 @@
+
+# This workflow runs pre-commit hooks on pushes and pull requests to main
+# to enforce coding style. To ensure correct configuration, please refer to:
+#  https://lincc-ppt.readthedocs.io/en/latest/practices/ci_precommit.html
+name: Run pre-commit hooks
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  pre-commit-ci:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v5
+      with:
+        fetch-depth: 0 
+    # - name: Set up Python
+    #   uses: actions/setup-python@v6
+    #   with:
+    #     python-version: '3.11'
+    # - name: Install dependencies
+    #   run: |
+    #     sudo apt-get update
+    #     python -m pip install --upgrade pip
+    #     pip install .[dev]
+    #     if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - uses: conda-incubator/setup-miniconda@v3
+      with:
+        python-version: ${{ matrix.pyver }}
+        channels: conda-forge
+        channel-priority: strict
+        show-channel-urls: true
+        miniforge-version: latest
+    - name: configure conda and install code
+      shell: bash -l {0}
+      run: |
+        conda config --set always_yes yes
+        conda install --quiet --yes --file requirements-conda.txt
+        python -m pip install -r requirements.txt
```

Comment:
I am not a big fan of having both conda and pip to manage dependencies. Is there a way to avoid pip?
I guess `pyimcom` is the problem here. Or maybe only keep pyimcom in the requirements?

Reply:
Yes, we need pip for pyimcom, and we need conda for mdet (due to dependencies on ngmix/esutil/meds).

The `requirements.txt` file is being used as the dependencies in `pyproject.toml`. In practice during installation, it will notice that nearly everything is already provided (after having setup the environment with conda)

Reply:
I'd love to move to `uv` but the conda requirements make that intractable for now :shrug:

Reply:
I completely agree. Ideally we would figure out how to get `metadetect` to install purely via pip, but I haven't had any success doing so